tests/integration/ha/test_network_cut.py

```diff
@@ -0,0 +1,426 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+from pathlib import Path
+from subprocess import CalledProcessError
+
+import integration.ha.helpers as ha_helpers
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+from ..helpers import access_all_dashboards, get_leader_name
+
+logger = logging.getLogger(__name__)
+
+
+CLIENT_TIMEOUT = 10
+RESTART_DELAY = 60
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+OPENSEARCH_APP_NAME = "opensearch"
+OPENSEARCH_CONFIG = {
+    "logging-config": "<root>=INFO;unit=DEBUG",
+    "cloudinit-userdata": """postruncmd:
+        - [ 'sysctl', '-w', 'vm.max_map_count=262144' ]
+        - [ 'sysctl', '-w', 'fs.file-max=1048576' ]
+        - [ 'sysctl', '-w', 'vm.swappiness=0' ]
+        - [ 'sysctl', '-w', 'net.ipv4.tcp_retries2=5' ]
+    """,
+}
+TLS_CERTIFICATES_APP_NAME = "self-signed-certificates"
+
+NUM_UNITS_APP = 2
+NUM_UNITS_DB = 3
+
+
+# @pytest.fixture()
+# async def restart_delay(ops_test: OpsTest):
+#     for unit in ops_test.model.applications[ha_helpers.APP_NAME].units:
+#         await ha_helpers.patch_restart_delay(
+#             ops_test=ops_test, unit_name=unit.name, delay=RESTART_DELAY
+#         )
+#     yield
+#     for unit in ops_test.model.applications[ha_helpers.APP_NAME].units:
+#         await ha_helpers.remove_restart_delay(ops_test=ops_test, unit_name=unit.name)
+#
+#
+# @pytest.fixture()
+# async def no_lxd_dnsmasq():
+#     ha_helpers.disable_lxd_dnsmasq()
+#     yield
+#     ha_helpers.enable_lxd_dnsmasq()
+
+
+async def extra_secure_wait_for_idle(ops_test: OpsTest):
+    """Particularly strong, specific wait_for_idle time period.
```

Comment:
We should move to the `wait_until`, as the opensearch itself is doing.

Reply:
This one seemed reliable for this particular case. https://github.com/juju/python-libjuju/blob/d4f667766655e1eadbb5fd8e6cb40eb9e9451039/juju/model.py#L2829-L2832

As I understand  `wait_for_at_least_units` isn't only checking for the number of those units, but also for the `idle` state of them. Pls correct if I'm wrong.

TBH this change I may prefer as a separate PR. The simple reason is that HA tests are _tedious_ and take a long time to debug.

Now we have a stable solution here. I'd perefer to have a "less fancy" stable thing, than perhaps a broken fancy one.

How about me addressint this in a separate PR, and we sort out all the rest on this one?

Reply:
I noticed you replaced parts with `block_until`. I think this is fine.